src/rust/engine/fs/src/lib.rs

```diff
@@ -687,7 +736,40 @@ impl PosixFS {
   }
 
   pub fn is_ignored(&self, stat: &Stat) -> bool {
-    self.ignore.is_ignored(stat)
+    println!("Are we actually invoking is_ignored? Stat: {:?}", stat);
+    if self.heed_gitignore_files {
```

Comment:
This method is going to be called for every path that is matched or walked, so I don't think that re-parsing the git-ignore in each case is going to be viable from a performance perspective.